init dev commit

slwtk opened the pull request:
added signup + jwt auth

slwtk committed:
init dev commit

egrva commented on pom.xml:
```diff
@@ -0,0 +1,96 @@
+    <name>notarize</name>
+    <description>Notarize project</description>
+    <properties>
+        <java.version>11</java.version>
```
нифига ахахаххахахаха 11 прикольно

egrva commented on src/main/java/ru/itis/notarize/controllers/TestController.java:
```diff
@@ -0,0 +1,36 @@
+
+@Slf4j
+@RestController
+@AllArgsConstructor
```
лучше заменить аннотацией @RequiredArgsConstructor

egrva commented on src/main/java/ru/itis/notarize/controllers/TestController.java:
```diff
@@ -0,0 +1,36 @@
+
+    @GetMapping("/user")
+    @PreAuthorize("hasRole('USER')")
+    public ResponseEntity<User> hello(@AuthenticationPrincipal String username) {
```
лучше отдавать dto класс и можно без обертки в ResponseEntity

egrva commented on src/main/java/ru/itis/notarize/controllers/TestController.java:
```diff
@@ -0,0 +1,36 @@
+    @GetMapping("/user")
+    @PreAuthorize("hasRole('USER')")
+    public ResponseEntity<User> hello(@AuthenticationPrincipal String username) {
+        return ResponseEntity.of(userRepository.findByUsername(username));
```
почему напрямую к репозиторию? такую логику лучше выносить в service

egrva commented on src/main/java/ru/itis/notarize/security/jwt/UsernameAndPasswordRequest.java:
```diff
@@ -0,0 +1,26 @@
+    private String password;
+
+
+    public String getUsername() {
```
ты же и так @Data повесил, зачем еще геттеры и стеттеры тут?

egrva commented on src/main/java/ru/itis/notarize/controllers/api/NotaryRestController.java:
```diff
@@ -0,0 +1,25 @@
+@AllArgsConstructor
+public class NotaryRestController {
+
+    private final UserRepository userRepository;
```
почему здесь userRepository? и запрос возвращает User? у тебя же кастомные сущности, расширяющие user

slwtk committed:
file upload + some refactoring

egrva commented on src/main/java/ru/itis/notarize/controllers/api/ClientRestController.java:
```diff
@@ -0,0 +1,42 @@
+
+    @GetMapping("/profile")
+    @PreAuthorize("hasRole('CLIENT')")
+    public ClientDto profile(@AuthenticationPrincipal String username) {
```
а че не поменял сигнатуру? ты же хотел нормально сделать без этой фигни секьюрной

egrva commented on src/main/java/ru/itis/notarize/controllers/api/ClientRestController.java:
```diff
@@ -0,0 +1,42 @@
+    }
+
+    @PatchMapping
+    public ResponseEntity<ClientDto> editClient(@AuthenticationPrincipal String username,
```
где-то убрал resposneEntity где-то оставил, так не пойдет

egrva commented on src/main/java/ru/itis/notarize/controllers/api/FileRestController.java:
```diff
@@ -0,0 +1,36 @@
+
+    private final FileService fileService;
+
+    public FileRestController(FileService fileService) {
```
с аннотацией красивей как по мне

slwtk commented:
Preparing review...

slwtk commented:
Preparing review...

slwtk commented:
Preparing review...

slwtk commented:
### **Ask**❓
Summarize the pull request

### **Answer:**
This pull request adds several new features and components to the Notarize application. The main changes include:

1. **New User Roles**: Introduces two new user roles, `ROLE_CLIENT` and `ROLE_NOTARY`, each with their respective DTOs (`ClientDto` and `NotaryDto`) and mappers (`ClientMapper`/`ClientMapperImpl` and `NotaryMapper`/`NotaryMapperImpl`).

2. **File Management**: Adds functionality to manage files, including uploading and retrieving file details. This includes a new model (`File`), DTO (`FileDto`), and mapper (`FileMapper`/`FileMapperImpl`).

3. **Password Encoding**: Utilizes Spring's `PasswordEncoder` for securely encoding user passwords.

4. **Role-Based Access Control (RBAC)**: Enhances the application to support role-based access control, ensuring that users with different roles have appropriate permissions.

5. **Configuration**: Updates the configuration files to include new properties and settings related to the new features.

6. **Testing**: Includes basic unit tests for the new components to ensure they function as expected.

The pull request also includes refactoring of existing code to integrate these new features seamlessly, such as updating service classes and controllers to handle new user roles and file management operations.

Overall, this PR significantly expands the functionality of the Notarize application by adding robust support for client and notary users, file management, and enhanced security through role-based access control.

slwtk commented:
### **Ask**❓
Summarize the pull request

### **Answer:**
This pull request introduces several new components and interfaces to enhance the functionality of a notarization application. The key changes include:

1. **New Models**: Introduced `Client`, `Notary`, and `File` models with their respective DTOs (`ClientDto`, `NotaryDto`, `FileDto`) and mappers (`ClientMapper`, `NotaryMapper`, `FileMapper`).

2. **Service Layer**: Added services for handling clients, notaries, and files.

3. **Repository Layer**: Created repositories for data persistence of clients, notaries, and files.

4. **Controller Layer**: Implemented controllers to handle HTTP requests for managing clients, notaries, and files.

5. **Security Configuration**: Enhanced security configuration to include user authentication and authorization.

6. **File Upload Handling**: Added functionality to upload and manage files securely.

7. **Testing**: Included unit tests for the new components and services.

The pull request also includes refactoring of existing code to adhere to best practices, such as using interfaces for mappers and repositories, and ensuring that all components are properly tested.

Overall, this PR significantly expands the functionality of the notarization application by adding features for managing clients, notaries, and files, along with improved security and file handling capabilities.

slwtk commented:
### **Ask**❓
Summarize the pull request

### **Answer:**
This pull request adds several new components and interfaces to the Notarize application. It includes DTOs (Data Transfer Objects), mappers, services, repositories, and controllers for handling clients, notaries, files, and other entities. The changes also include updates to the configuration and security settings to support these new features.

The PR aims to enhance the functionality of the Notarize application by providing a more structured and modular approach to managing different types of users and their associated data.

slwtk commented:
Preparing review...

slwtk commented:
Preparing PR description...

slwtk commented:
Preparing review...

slwtk commented:
Preparing PR description...

slwtk commented:
Preparing PR description...

slwtk commented:
Preparing PR description...

slwtk commented:
Preparing PR description...

slwtk commented:
Preparing PR description...

slwtk commented:
Preparing PR description...

slwtk commented:
Preparing PR description...

slwtk commented:
Preparing PR description...

slwtk commented:
Preparing PR description...

slwtk commented:
Preparing PR description...
